driver/controller.go

```diff
@@ -169,6 +169,7 @@ func (d *Driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest)
 			if resp != nil && resp.StatusCode == http.StatusNotFound {
 				return nil, status.Errorf(codes.NotFound, "snapshot %q not found", snapshotID)
 			}
+			return nil, err
```

Comment:
We do not seem to return a proper gRPC status code in similar code paths either, so I copied the pattern. Not sure if good or not, my expectation would be that gRPC maps this to a generic "internal error" or similar.

@adamwg you may know this one better than me.

Reply:
I believe the go gRPC framework wraps non-gRPC errors from RPC handlers using the `Unknown` status code. I always prefer to return explicit gRPC errors with appropriate codes from handlers, but gRPC is OK with it either way.

Makes sense to follow the general pattern here, and if we want to return explicit `Internal`s everywhere we can do that in a separate RP.

Reply:
Makes sense to me -- will keep that in mind for a follow up PR.